eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/processors/SubmittedVariantDeclusterProcessorTest.java

```diff
@@ -0,0 +1,192 @@
+                                                          VariantType.DEL);
+        processor.process(wrapper);
+
+        //Check Decluster
+        assertEquals(3, wrapper.getSubmittedVariants().size());
+        assertNotNull(wrapper.getSubmittedVariants().get(0).getClusteredVariantAccession());
+        assertNull(wrapper.getSubmittedVariants().get(1).getClusteredVariantAccession());
+        assertNull(wrapper.getSubmittedVariants().get(2).getClusteredVariantAccession());
+
+        //Check Operations
+        assertEquals(2, wrapper.getOperations().size());
+        assertOperation(wrapper, getReason(Collections.singletonList(DECLUSTERED_TYPE_MISMATCH)),
+                         wrapper.getSubmittedVariants().get(0).getAccession(), 0);
+        assertOperation(wrapper, getReason(Collections.singletonList(DECLUSTERED_TYPE_MISMATCH)),
+                         wrapper.getSubmittedVariants().get(1).getAccession(), 1);
+    }
+
+    @Test
+    public void processSubmittedVariantAllelesAndTypeMismatch() throws Exception {
+        List<DbsnpSubmittedVariantEntity> submittedVariantEntities = createSubmittedVariantEntities("G", "-/T/TT",
+                                                                                                    false);
+        DbsnpVariantsWrapper wrapper = buildSimpleWrapper(submittedVariantEntities, DbsnpVariantType.DIV,
+                                                          VariantType.DEL);
+        processor.process(wrapper);
+
+        //Check Decluster
+        assertEquals(3, wrapper.getSubmittedVariants().size());
+        assertNull(wrapper.getSubmittedVariants().get(0).getClusteredVariantAccession());
+        assertNull(wrapper.getSubmittedVariants().get(1).getClusteredVariantAccession());
+        assertNull(wrapper.getSubmittedVariants().get(2).getClusteredVariantAccession());
+
+        //Check Operations
+        assertEquals(3, wrapper.getOperations().size());
+        assertOperation(wrapper, getReason(Collections.singletonList(DECLUSTERED_ALLELES_MISMATCH)),
+                        wrapper.getSubmittedVariants().get(0).getAccession(), 0);
+        assertOperation(wrapper, getReason(Arrays.asList(DECLUSTERED_ALLELES_MISMATCH, DECLUSTERED_TYPE_MISMATCH)),
+                        wrapper.getSubmittedVariants().get(1).getAccession(), 1);
+        assertOperation(wrapper, getReason(Arrays.asList(DECLUSTERED_ALLELES_MISMATCH, DECLUSTERED_TYPE_MISMATCH)),
+                        wrapper.getSubmittedVariants().get(2).getAccession(), 2);
+    }
+
+    private List<DbsnpSubmittedVariantEntity> createSubmittedVariantEntities(String reference, String alternate,
+                                                                             boolean allelesMatch) {
+        List<DbsnpSubmittedVariantEntity> submittedVariantEntities = new ArrayList<>();
+        String[] alternates = Arrays.stream(StringUtils.split(alternate, "/")).toArray(String[]::new);
```

Comment:
It would be easier to make `alternate` into an array argument instead of a string, as in `new String[]{ "-", "T", "TT" }`. It reduces a bit the copy & paste possibilities, but there aren't many of these test cases anyway.